### PR TITLE
New features related to MEGAchat

### PR DESCRIPTION
**Pull request by sergiohs84**

### CODE REVIEW

File: include/megaapi.h [hunk 1]
```diff
@@ -9004,6 +9005,61 @@ class MegaApi
...
+         *
+         * Valid data in the MegaRequest object received in onRequestFinish when the error code
+         * is MegaError::API_OK:
+         * - MegaRequest::getNumber - Return the HTTP status code from the GeLB server
+         * - MegaRequest::getText - Returns the JSON response from the GeLB server
+         * - MegaRequest::getTotalBytes - Returns the number of bytes in the response
+         *
+         * @param service Service to check
+         * @param timeoutms Timeout for the request, including all possible retries
+         * @param listener MegaRequestListener to track this request
```
Comment (polmr) [inline include/megaapi.h]: missing @param maxretries

File: src/megaclient.cpp [hunk 2]
```diff
@@ -993,6 +999,73 @@ void MegaClient::exec()
...
+                                req->post(this);
+                                break;
+                            case METHOD_NONE:
+                                req->dns(this);
+                                break;
+                        }
+                        it++;
+                        break;
+                    }
+                    // no retry -> fall through
```
Comment (polmr) [inline src/megaclient.cpp]: missing it++ & break here?

Comment (javiserrano) [inline src/megaclient.cpp]: It is on purpose to check if the maximum time available for the request has expired during the time between retries

File: bindings/ios/3rdparty/different_address.patch [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+-    }
+-
+   if (channel->sock_create_cb)
+     {
+       int err = channel->sock_create_cb(s, SOCK_DGRAM,
+@@ -1244,34 +1227,6 @@
+   return 1;
+ }
+ 
+-static int same_address(struct sockaddr *sa, struct ares_addr *aa)
```
Comment (javiserrano) [inline bindings/ios/3rdparty/different_address.patch]: I agree, but it helps to check that it's not used elsewhere.

File: include/megaapi.h (hunk 1 above)
Comment (javiserrano) [inline include/megaapi.h]: I'm going to add it.

**Pull request merged**